src/Reflection/ReflectionClass.php

```diff
@@ -236,7 +238,16 @@ public static function createFromNode(
      */
     public function getShortName() : string
     {
-        return $this->node->name;
+        if (!$this->isAnonymous()) {
+            return $this->node->name;
+        }
+
+        $fileName = $this->getFileName();
+        if ($fileName === null) {
```

Comment:
Yes but I need new NodeVisitor to find `Eval_` and its content. Then I need the parse the content and find ast of the anonymous class.

And there could be situations like two `Eval_` on same line, two anonymous classes in the `Eval_` content...